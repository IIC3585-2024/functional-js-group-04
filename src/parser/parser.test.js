import parse from "./parser";

describe("parser", () => {
  const prefix =
    "<!DOCTYPE html><html><head><title>Markdown</title></head><body>";
  const suffix = "</body></html>";

  test("should return a function", () => {
    expect(typeof parse).toBe("function");
  });

  test("should support heading tokens", () => {
    const tokens = [
      {
        children: [
          {
            raw: "foo ",
            text: "foo ",
            type: "text_inline",
          },
          {
            raw: "**bar**",
            text: "bar",
            type: "bold",
          },
          {
            raw: " baz",
            text: " baz",
            type: "text_inline",
          },
        ],
        level: 1,
        raw: "# foo **bar** baz",
        text: "foo **bar** baz",
        type: "heading",
      },
    ];

    expect(parse(tokens)).toMatchSnapshot();
  });

  test("should support text blocks with inline content", () => {
      const tokens = [
        {
          "children": [
            {
              "raw": "foo ",
              "text": "foo ",
              "type": "text_inline",
            },
            {
              "raw": "**bar**",
              "text": "bar",
              "type": "bold",
            },
            {
              "raw": " baz ",
              "text": " baz ",
              "type": "text_inline",
            },
            {
              "raw": "*qux*",
              "text": "qux",
              "type": "italic",
            },
          ],
          "raw": "foo **bar** baz *qux*",
          "text": "foo **bar** baz *qux*",
          "type": "text_block",
        },
      ];

    expect(parse(tokens)).toMatchSnapshot();
  });
});
