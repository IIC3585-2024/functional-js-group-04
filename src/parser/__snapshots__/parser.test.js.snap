// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser spec with fenced code block 1`] = `
"<!DOCTYPE html><style>undefined</style><html><head><title>Markdown</title></head><body><pre><code>function foo(x) {
  return 3
}</code></pre></body></html>"
`;

exports[`parser spec with heading tokens 1`] = `"<!DOCTYPE html><style>undefined</style><html><head><title>Markdown</title></head><body><h1>foo <strong>bar</strong> baz</h1></body></html>"`;

exports[`parser spec with indented code block 1`] = `
"<!DOCTYPE html><style>undefined</style><html><head><title>Markdown</title></head><body><pre><code>a simple
  indented code block</code></pre></body></html>"
`;

exports[`parser spec with paragraphs 1`] = `
"<!DOCTYPE html><style>undefined</style><html><head><title>Markdown</title></head><body><p>foo
bar
baz</p></body></html>"
`;

exports[`parser spec with thematic break 1`] = `"<!DOCTYPE html><style>undefined</style><html><head><title>Markdown</title></head><body><hr/><hr/><hr/></body></html>"`;
